galasa-managers-parent/galasa-managers-core-parent/dev.galasa.core.manager/src/main/java/dev/galasa/core/manager/internal/ResourceStringGenerator.java

```diff
@@ -0,0 +1,200 @@
+/*
+* Copyright contributors to the Galasa project 
+*/
+package dev.galasa.core.manager.internal;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map.Entry;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import dev.galasa.core.manager.CoreManagerException;
+import dev.galasa.core.manager.ResourceString;
+import dev.galasa.core.manager.internal.properties.ResourceStringPattern;
+import dev.galasa.framework.spi.DssAdd;
+import dev.galasa.framework.spi.DssDelete;
+import dev.galasa.framework.spi.DynamicStatusStoreException;
+import dev.galasa.framework.spi.DynamicStatusStoreMatchException;
+import dev.galasa.framework.spi.IDynamicStatusStoreService;
+import dev.galasa.framework.spi.IResourcePoolingService;
+import dev.galasa.framework.spi.InsufficientResourcesAvailableException;
+import dev.galasa.framework.spi.ResourceUnavailableException;
+
+/**
+ * Generate Resource Strings using random letters
+ * 
+ * 
+ * @author Michael Baylis
+ *
+ */
+public class ResourceStringGenerator {
+
+	private final static Log                    logger = LogFactory.getLog(ResourceStringGenerator.class);
+
+	private CoreManagerImpl                     coreManager;
+	// Maintain a map of what tags have already been provisioned so duplicates are not created
+	// Also maintain a list so the Resource Strings can be discarded at the end of the run
+	private HashMap<String, ResourceStringImpl> resourceStrings = new HashMap<>();
+
+
+	public ResourceStringGenerator(CoreManagerImpl coreManager)   {
+		this.coreManager = coreManager;
+	}
+
+	/**
+	 * Attempt to generate a string for a field
+	 * 
+	 * @param field - The field to provision for
+	 * @return A resource string implementation
+	 * @throws CoreManagerException
+	 * @throws ResourceUnavailableException
+	 */
+	public ResourceStringImpl generateString(Field field) throws CoreManagerException, ResourceUnavailableException {
+		// Get the @ResourceString annotation, with the length and tag
+		ResourceString annotation = field.getAnnotation(ResourceString.class);
+		int stringLength = annotation.length();
```

Comment:
Have to admit, not how i thought this was going to work when i looked at the @interface. I was expecting it to be a provide a tag to reference a char definition string in the CPS OR provide a length to get given a {a-zA-Z0-9}{a-zA-Z0-9}{a-zA-Z0-9}...{a-zA-Z0-9}. Also we are going to encounter issues here if people require two resource strings and dont provide a tag as both will get set to PRIMARY and as they will override one another in the resourceStrings map. Would be forcing the resourceMonitor to clean it up, which is less than ideal